gui/src/app/SamplerOutputView/SamplerOutputView.tsx

```diff
@@ -197,6 +218,29 @@ const prepareCsvText = (draws: number[][], paramNames: string[], drawChainIds: n
     return [['Chain', 'Draw', ...paramNames].join(','), ...lines].join('\n')
 }
 
+const prepareMultipleCsvsText = (draws: number[][], paramNames: string[], drawChainIds: number[], uniqueChainIds: number[]) => {
+    return uniqueChainIds.map(chainId => {
+        const drawIndicesForChain = drawChainIds.map((id, i) => id === chainId ? i : -1).filter(i => i >= 0);
+        const lines = drawIndicesForChain.map(i => {
+            return paramNames.map((_, j) => draws[j][i]).join(',')
+        })
+
+        return [paramNames.join(','), ...lines].join('\n')
+    })
+}
+
+const createZipBlobForMultipleCsvs = async (csvTexts: string[], uniqueChainIds: number[]) => {
+    const zip = new JSZip();
+    // put them all in a folder called 'draws'
+    const folder = zip.folder('draws');
```

Comment:
This is really a separate feature request, but it would be nice if we included samplerOpts.json in this download as well, for reference. Not necessary in V1 on this, since I think that would make it depend on some other PRs

Reply:
created  new issue.